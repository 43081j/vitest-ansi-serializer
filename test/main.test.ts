import {test, expect, suite, SnapshotSerializer} from 'vitest';
import ansiSerializer from '../src/main.js';
import {cursor, erase, scroll} from 'sisteransi';

type NewSnapshotSerializer = Exclude<SnapshotSerializer, {print: unknown}>;

const serializer = ansiSerializer as NewSnapshotSerializer;

const mockConfig: Parameters<NewSnapshotSerializer['serialize']>[1] =
  {} as never;
const toString = (val: unknown) => String(val);
const basicSerialize = (input: unknown) =>
  serializer.serialize(input, mockConfig, '', 0, [], toString);

const serializeCases: Array<[name: string, input: string]> = [
  ['cursor.backward()', `foo${cursor.backward()}`],
  ['cursor.backward(n)', `foo${cursor.backward(2)}`],
  ['cursor.down()', `foo${cursor.down()}`],
  ['cursor.down(n)', `foo${cursor.down(2)}`],
  ['cursor.forward()', `foo${cursor.forward()}`],
  ['cursor.forward(n)', `foo${cursor.forward(2)}`],
  ['cursor.hide', `foo${cursor.hide}`],
  ['cursor.left', `foo${cursor.left}`],
  ['cursor.move(1, 2)', `foo${cursor.move(1, 2)}`],
  ['cursor.nextLine()', `foo${cursor.nextLine()}`],
  ['cursor.nextLine(n)', `foo${cursor.nextLine(2)}`],
  ['cursor.prevLine()', `foo${cursor.prevLine()}`],
  ['cursor.prevLine(n)', `foo${cursor.prevLine(2)}`],
  ['cursor.restore', `foo${cursor.restore}`],
  ['cursor.save', `foo${cursor.save}`],
  ['cursor.show', `foo${cursor.show}`],
  ['cursor.to(1, 2)', `foo${cursor.to(1, 2)}`],
  ['cursor.up', `foo${cursor.up()}`],
  ['cursor.up(n)', `foo${cursor.up(2)}`],
  ['erase.down(1)', `foo${erase.down(1)}`],
  ['erase.line', `foo${erase.line}`],
  ['erase.lineEnd', `foo${erase.lineEnd}`],
  ['erase.lineStart', `foo${erase.lineStart}`],
  ['erase.lines(1)', `foo${erase.lines(1)}`],
  ['erase.lines(10)', `foo${erase.lines(10)}`],
  ['erase.screen', `foo${erase.screen}`],
  ['erase.up()', `foo${erase.up()}`],
  ['erase.up(n)', `foo${erase.up(2)}`],
  ['scroll.down()', `foo${scroll.down()}`],
  ['scroll.down(n)', `foo${scroll.down(2)}`],
  ['scroll.up()', `foo${scroll.up()}`],
  ['scroll.up(n)', `foo${scroll.up(2)}`],
  ['multiple cursor movements', `foo${cursor.up(3)}bar${cursor.backward(10)}`]
];
suite('serializer', () => {
  suite('serialize', () => {
    test('returns non-ansi string unchanged', () => {
      const input = 'choo choo';
      expect(basicSerialize(input)).toMatchSnapshot();
    });

    test.for(serializeCases)('%s', ([_name, input]) => {
      expect(basicSerialize(input)).toMatchSnapshot();
    });
  });

  suite('test', () => {
    test('returns true for string', () => {
      const result = serializer.test('string');
      expect(result).toBe(true);
    });

    test('returns false for non-string', () => {
      const result = serializer.test(123);
      expect(result).toBe(false);
    });
  });
});
