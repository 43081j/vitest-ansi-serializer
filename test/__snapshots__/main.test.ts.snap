// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`serializer > serialize > cursor.backward() 1`] = `"foo<cursor.backward count=1>"`;

exports[`serializer > serialize > cursor.backward(n) 1`] = `"foo<cursor.backward count=2>"`;

exports[`serializer > serialize > cursor.down() 1`] = `"foo<cursor.down count=1>"`;

exports[`serializer > serialize > cursor.down(n) 1`] = `"foo<cursor.down count=2>"`;

exports[`serializer > serialize > cursor.forward() 1`] = `"foo<cursor.forward count=1>"`;

exports[`serializer > serialize > cursor.forward(n) 1`] = `"foo<cursor.forward count=2>"`;

exports[`serializer > serialize > cursor.hide 1`] = `"foo<cursor.hide>"`;

exports[`serializer > serialize > cursor.left 1`] = `"foo<cursor.left count=1>"`;

exports[`serializer > serialize > cursor.move(1, 2) 1`] = `"foo<cursor.forward count=1><cursor.down count=2>"`;

exports[`serializer > serialize > cursor.nextLine() 1`] = `"foo<cursor.nextLine count=1>"`;

exports[`serializer > serialize > cursor.nextLine(n) 1`] = `"foo<cursor.nextLine count=1><cursor.nextLine count=1>"`;

exports[`serializer > serialize > cursor.prevLine() 1`] = `"foo<cursor.prevLine count=1>"`;

exports[`serializer > serialize > cursor.prevLine(n) 1`] = `"foo<cursor.prevLine count=1><cursor.prevLine count=1>"`;

exports[`serializer > serialize > cursor.restore 1`] = `"foo<cursor.restore>"`;

exports[`serializer > serialize > cursor.save 1`] = `"foo<cursor.save>"`;

exports[`serializer > serialize > cursor.show 1`] = `"foo<cursor.show>"`;

exports[`serializer > serialize > cursor.to(1, 2) 1`] = `"foo<cursor.moveTo line=3 column=2>"`;

exports[`serializer > serialize > cursor.up 1`] = `"foo<cursor.up count=1>"`;

exports[`serializer > serialize > cursor.up(n) 1`] = `"foo<cursor.up count=2>"`;

exports[`serializer > serialize > erase.down(1) 1`] = `"foo<erase.down>"`;

exports[`serializer > serialize > erase.line 1`] = `"foo<erase.line>"`;

exports[`serializer > serialize > erase.lineEnd 1`] = `"foo<erase.lineEnd>"`;

exports[`serializer > serialize > erase.lineStart 1`] = `"foo<erase.lineStart>"`;

exports[`serializer > serialize > erase.lines(1) 1`] = `"foo<erase.line><cursor.left count=1>"`;

exports[`serializer > serialize > erase.lines(10) 1`] = `"foo<erase.line><cursor.up count=1><erase.line><cursor.up count=1><erase.line><cursor.up count=1><erase.line><cursor.up count=1><erase.line><cursor.up count=1><erase.line><cursor.up count=1><erase.line><cursor.up count=1><erase.line><cursor.up count=1><erase.line><cursor.up count=1><erase.line><cursor.left count=1>"`;

exports[`serializer > serialize > erase.screen 1`] = `"foo<erase.screen>"`;

exports[`serializer > serialize > erase.up() 1`] = `"foo<erase.up>"`;

exports[`serializer > serialize > erase.up(n) 1`] = `"foo<erase.up><erase.up>"`;

exports[`serializer > serialize > multiple cursor movements 1`] = `"foo<cursor.up count=3>bar<cursor.backward count=10>"`;

exports[`serializer > serialize > returns non-ansi string unchanged 1`] = `"choo choo"`;

exports[`serializer > serialize > scroll.down() 1`] = `"foo<cursor.scrollDown count=1>"`;

exports[`serializer > serialize > scroll.down(n) 1`] = `"foo<cursor.scrollDown count=1><cursor.scrollDown count=1>"`;

exports[`serializer > serialize > scroll.up() 1`] = `"foo<cursor.scrollUp count=1>"`;

exports[`serializer > serialize > scroll.up(n) 1`] = `"foo<cursor.scrollUp count=1><cursor.scrollUp count=1>"`;
